airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -114,12 +116,12 @@ def _get_authenticator(config: Dict[str, Any]):
         return MultipleTokenAuthenticator(tokens=tokens, auth_method="token")
 
     @staticmethod
-    def _get_branches_data(selected_branches: str, full_refresh_args: Dict[str, Any] = None) -> Tuple[Dict[str, str], Dict[str, List[str]]]:
+    def _get_branches_data(selected_branches: str, config: Mapping[str, Any], full_refresh_args: Dict[str, Any] = None) -> Tuple[Dict[str, str], Dict[str, List[str]]]:
         selected_branches = set(filter(None, selected_branches.split(" ")))
 
         # Get the default branch for each repository
         default_branches = {}
-        repository_stats_stream = RepositoryStats(**full_refresh_args)
+        repository_stats_stream = RepositoryStats(config, **full_refresh_args)
```

Comment:
I don't think it is correct to pass the complete `config` object to each stream. Please create a new variable and validate if the `config.get("api_url")` exists if not use a default value to not have backward compatibility issues.